File: CS/WindowsApplication1/Form1.cs
using DevExpress.XtraReports.UI;
using System;
using System.Windows.Forms;

namespace WindowsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        private void button1_Click(object sender, EventArgs e) {
            InitializeReport();
        }
        private void InitializeReport() {
            XtraReport1 report = new XtraReport1();
            report.DataSource = FillReportDataSource();
            ReportPrintTool tool = new ReportPrintTool(report);
            tool.ShowPreviewDialog();
        }
        object FillReportDataSource() {
            return new DemoDataSource().GetData();
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Windows.Forms

Namespace WindowsApplication1
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub ShowReport()
            Dim report As XtraReport1 = New XtraReport1()
            report.DataSource = FillReportDataSource()
            Dim tool As ReportPrintTool = New ReportPrintTool(report)
            tool.ShowPreviewDialog()
        End Sub

        Private Function FillReportDataSource() As Object
            Return New DemoDataSource().GetData()
        End Function

        Private Sub button1_Click_1(sender As Object, e As EventArgs) Handles button1.Click
            ShowReport()
        End Sub
    End Class
End Namespace
